tests/ResultTests.cpp

```diff
@@ -578,6 +578,20 @@ TEST_CASE("ResultTests::AutomaticOriginationOnFailure", "[result]")
     REQUIRE(S_FALSE == GetRestrictedErrorInfo(&restrictedErrorInformation));
 }
 
+void ExpectRestrictedError(HRESULT err, LPCWSTR text)
+{
+    wil::com_ptr_nothrow<IRestrictedErrorInfo> errorInfo;
+    REQUIRE_SUCCEEDED(GetRestrictedErrorInfo(&errorInfo));

```

Comment:
Perhaps worth a comment regarding the (extremely) nonobvious behavior that calling `GetRestrictedErrorInfo` removes the error information for the current thread.  That leaves it clean for the next test case.